tests/integration/targets/mso_backup/tasks/main.yml

```diff
@@ -43,540 +44,503 @@
+
+- name: Restore backup check
+  when: version.current.version is version('2.2.4e', '!=')
+  block:
+    - name: Restore non existent backup
+      mso_backup:
+        <<: *mso_info
+        backup: non_existent_backup
+        state: restore
+        timeout: 1000
+      ignore_errors: yes
+      register: restore_non_existent_backup
+
+    - name: Add a new tenant
+      mso_tenant:
+        <<: *tenant1_absent
+        state: present
+      register: tenant1_present
+
+    - name: Restore {{ uploaded_backup_name }} in check mode
+      mso_backup: &restore_backup_cm
+        <<: *mso_info
+        backup: '{{ uploaded_backup_name }}'
+        state: restore
+        timeout: 1000
+      check_mode: yes
+      register: restore_backup_cm
+
+    - name: Restore {{ uploaded_backup_name }} in normal mode
+      mso_backup:
+        <<: *restore_backup_cm
+      register: restore_backup_nm
+
+    - name: Query Tenant1 after restoring the backup
+      mso_tenant:
+        <<: *tenant1_absent
+        state: query
+      register: query_tenant1
+
+    - name: Restore {{ uploaded_backup_name }} in normal mode - idempotency check
+      mso_backup:
+        <<: *restore_backup_cm
+      register: idm_restore_backup_nm
+
+    - name: Assertions check for the backup restore
+      assert:
+        that:
+          - restore_non_existent_backup.msg is match ("Backup 'non_existent_backup' does not exist")
+          - restore_backup_cm is changed
+          - restore_backup_nm is changed
+          - idm_restore_backup_nm is changed
+          - tenant1_present.current != {}
+          - query_tenant1 is not changed
+          - query_tenant1.current == {}
+
+# Cleaning part
+# Remove all other backups
+- name: Query all backups before deleting
+  mso_backup:
+    <<: *query_all_backups
+    state: query
+  register: query_all_backups_bf_delete
+
+- name: Ensure backups does not exists
+  mso_backup:
+    <<: *mso_info
+    backup_id:  '{{ item.id }}'
+    state: absent
+  loop: '{{ query_all_backups_bf_delete.current | list | sort(attribute="name", reverse=True) }}'
+
+- name: Query all backups after deleting - to ensure all are absent
+  mso_backup:
+    <<: *query_all_backups
+  register: query_all_backups_af_delete
+
+- name: Assertions check for the query and remove all backups
+  assert:
+    that:
+      - query_all_backups_bf_delete is not changed
+      - query_all_backups_af_delete is not changed
+      - query_all_backups_bf_delete.current | selectattr("name", "match", "^ansibleBackup.*") | list | length != 0
+      - query_all_backups_af_delete.current | selectattr("name", "match", "^ansibleBackup.*") | list | length == 0
+
+# Clear Remote Location
+- name: Ensure remote location is absent
+  cisco.mso.mso_remote_location:
+    <<: *remote_location_absent
+
+# Remove Tenant1
+- name: Ensure Tenant1 is absent
+  mso_tenant:
+    <<: *tenant1_absent
+    state: absent
```

Comment:
why are few test cases removed from this file?

Reply:
I just merged and simplified the test cases to support MSO and NDO.